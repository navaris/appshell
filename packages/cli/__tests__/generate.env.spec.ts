import fs from 'fs';
import path from 'path';
import { rimrafSync } from 'rimraf';
import generateEnvHandler from '../src/handlers/generate.env';

describe('generate.env', () => {
  const prefix = '';
  const env = 'packages/cli/__tests__/assets/test.env';
  const outDir = path.resolve('packages/cli/__tests__/assets/temp');
  const outFile = 'appshell.env.js';
  const globalName = 'appshell_env';
  const overwrite = true;

  afterEach(() => {
    rimrafSync(outDir);
  });

  it('should generate the runtime environment js file', async () => {
    await generateEnvHandler({
      env,
      prefix,
      outDir,
      outFile,
      globalName,
      overwrite,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.toString()).toMatchSnapshot();
  });

  it('should capture only prefixed environment vars when prefix is supplied', async () => {
    await generateEnvHandler({
      env,
      prefix: 'TEST_',
      outDir,
      outFile,
      globalName,
      overwrite,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.toString()).toMatchSnapshot();
  });

  it('should output to outDir when supplied', async () => {
    const testPath = path.resolve('packages/cli/__tests__/assets/temp/test');
    await generateEnvHandler({
      env,
      prefix,
      outDir: testPath,
      outFile,
      globalName,
      overwrite,
    });
    const actual = fs.readFileSync(path.join(testPath, outFile));

    expect(actual.toString()).toBeTruthy();
  });

  it('should output with outFile name when supplied', async () => {
    const filename = 'test.env.js';
    await generateEnvHandler({
      env,
      prefix,
      outDir,
      outFile: filename,
      globalName,
      overwrite,
    });
    const actual = fs.readFileSync(path.join(outDir, filename));

    expect(actual.toString()).toBeTruthy();
  });

  it('should output with global variable name supplied', async () => {
    const testGlobalName = 'my_global_var';
    await generateEnvHandler({
      env,
      prefix,
      outDir,
      outFile,
      globalName: testGlobalName,
      overwrite,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.includes(`window.${testGlobalName}`)).toBe(true);
  });

  it('should reject when .env file does not exist', async () => {
    await expect(
      generateEnvHandler({
        env: 'assets/does_not_exist.env',
        prefix,
        outDir,
        outFile: 'appshell.env.js',
        globalName,
        overwrite,
      }),
    ).rejects.toThrow();
  });
});
