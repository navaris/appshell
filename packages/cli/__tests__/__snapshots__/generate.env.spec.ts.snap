// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate.env should capture only prefixed environment vars when prefix is supplied 1`] = `
"window.appshell_env = {
	APPSHELL_CONFIGS_DIR: "packages/cli/__tests__/assets/appshell_configs",
	APPSHELL_ROOT: "TestModule/Workspace",
}"
`;

exports[`generate.env should generate the runtime environment js file 1`] = `
"window.appshell_env = {
	APPSHELL_CONFIGS_DIR: "packages/cli/__tests__/assets/appshell_configs",
	APPSHELL_ROOT: "TestModule/Workspace",
	TEST_ENV_FOO: "foo",
	TEST_ENV_BAR: "bar",
}"
`;
